src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts

```diff
@@ -236,18 +240,41 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
   getGeneProperty(gene: GCTGene, property: string) {
     return property.split('.').reduce((o: any, i: any) => o[i], gene);
   }
-
+  
   initData(genes: GCTGene[]) {
+    // hide brain region columns initially
     this.brainRegionsColumns.forEach(c => c.visible = false);
-
+    
     const pinnedGenes: GCTGene[] = [];
-    const currentPinnedGenesCache = this.getPinnedGenesCache(
-      this.category,
-      this.subCategory
-    );
-    const urlPins = currentPinnedGenesCache.length
-      ? currentPinnedGenesCache.map((g: GCTGene) => g.uid)
-      : this.getUrlParam('pinned', true);
+    
+    const currentPinnedGenesCache = this.pinnedGenesCache;
+
+    let previousPins: (string | undefined)[] = [];
+
+    if (this.currentCategoriesMatchLastPinnedCategories()) {
+      // no need to load from cache if it was last pinned from the current page
+      // previousPins = this.getUrlParam('pinned', true);
+      previousPins = currentPinnedGenesCache.map((g: GCTGene) => g.uid);
```

Comment:
The pins are currently pulled from the cache, but the comment (and commented-out code) suggests that the pins will be pulled from the URL. Should this be updated to align the comment and the code?

Reply:
```suggestion
      previousPins = currentPinnedGenesCache.map((g: GCTGene) => g.uid);
```

Reply:
Removing the comment, since the if statement makes it pretty clear!